src/Kriti/Parser.hs

```diff
@@ -137,6 +145,32 @@ stringLit = match \case
   Lex.StringLit s -> Just s
   _               -> Nothing
 
+stringTem :: Parser [Either Text Lex.TokenStream]
+stringTem = do
+  res <- match \case
+    Lex.StringTem s -> Just $ traverse (traverse Lex.lexer) $ splitText s
+    _               -> Nothing
+  case res of
+    Left err  -> P.customFailure err
+    Right tem -> pure tem
+
+splitText :: Text -> [Either Text Text]
+splitText t = reverse $ go t []
```

Comment:
This function concerns me quite a bit. 

If we were to use Megaparsec to do this we would need an entirely seperate ParsecT type for consuming Text values vs our main one which uses our custom `TokenStream`. I believe this would lead to a complicated situation with regestering errors from this sub parser with our main parser.

Assuming we don't want to use Megaparsec here. Do you have any suggestions to simplify this function?